core/vdbe/mod.rs

```diff
@@ -3422,6 +3429,37 @@ fn exec_cast(value: &OwnedValue, datatype: &str) -> OwnedValue {
     }
 }
 
+fn exec_replace(source: &OwnedValue, pattern: &OwnedValue, replacement: &OwnedValue) -> OwnedValue {
+    // The replace(X,Y,Z) function returns a string formed by substituting string Z for every occurrence of
+    // string Y in string X. The BINARY collating sequence is used for comparisons. If Y is an empty string
+    // then return X unchanged. If Z is not initially a string, it is cast to a UTF-8 string prior to processing.
+
+    // If any of the arguments is NULL, the result is NULL.
+    if matches!(source, OwnedValue::Null)
+        || matches!(pattern, OwnedValue::Null)
+        || matches!(replacement, OwnedValue::Null)
+    {
+        return OwnedValue::Null;
+    }
+
+    let source = exec_cast(source, "TEXT");
+    let pattern = exec_cast(pattern, "TEXT");
+    let replacement = exec_cast(replacement, "TEXT");
+
+    // If any of the casts failed, return NULL
```

Comment:
looks like we are panicing instead of returning null, which is probably fine, but pls change this comment

Reply:
Fixed.